### PR TITLE
Fix undeclared identifier when HAVE_READLINK is not defined

### PR DESCRIPTION
**Pull request by mgnsk**
I was able to build neovim from master on Android/Termux (with the help of some patches found on reddit) with this fix.
I'm not even sure how it came up since I have `readlink` installed but still got the undeclared identifier error.

### CODE REVIEW

**Comment by jamessan** [comment]
> I'm not even sure how it came up since I have `readlink` installed but still got the undeclared identifier error.

`HAVE_READLINK` is related to the C function, not the shell command.

Thanks for the fix.  Looks good!

File: src/nvim/eval/funcs.c [hunk 1]
```diff
@@ -7044,7 +7044,7 @@ static void f_resolve(typval_T *argvars, typval_T *rettv, FunPtr fptr)
     xfree(buf);
   }
 # else
-  rettv->vval.v_string = (char_u *)xstrdup(p);
```
Comment (jamessan) [inline src/nvim/eval/funcs.c:7047]: Might be better to try using `os_realpath()` first (sort of like in the WIN32 case) and then fallback to just using the passed in value.

Comment (mgnsk) [reply src/nvim/eval/funcs.c:7047]: Gonna have to try what's available, it's very much a non-standard system.

Comment (jamessan) [reply src/nvim/eval/funcs.c:7047]: I mean something like:

```c
char *v = os_realpath(fname);
rettv->vval.v_string = (char_u *)(v == NULL ? xstrdup(fname) : v);
```

That way we at least give libuv a chance to help us out.

Comment (mgnsk) [reply src/nvim/eval/funcs.c:7047]: I tried out the `HAVE_READLINK` part and `os_realpath` separately, both seem to work. Termux also reports `-- Looking for readlink - found`.

Comment (jamessan) [reply src/nvim/eval/funcs.c:7047]: >  Termux also reports `-- Looking for readlink - found`.

Hmm, if that's the case and you're still getting into this part of the code, then something is wrong.  What does `grep READLINK build/config/auto/config.h` show?

Comment (mgnsk) [reply src/nvim/eval/funcs.c:7047]: Everything looks good:
```
$ grep READLINK build/config/auto/config.h
#define HAVE_READLINK
```
I'm also not able to reproduce the original error anymore - the latest `master` branch builds fine. Have no idea what's going on anymore.

Comment (jamessan) [reply src/nvim/eval/funcs.c:7047]: Ok, well either way this is an obvious fix for places where `readlink` isn't available.

**Pull request merged**